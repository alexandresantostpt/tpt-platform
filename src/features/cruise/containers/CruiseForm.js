import { connect } from 'react-redux'

import { getObjSelector as travelGetObjSelector, getClientsSelector } from '@features/travelScript/selectors'

import { getObjSelector } from '../selectors'
import { requestDelAction, requestGetSingleAction, requestSaveDataAction } from '../actions'

import CruiseForm from '../pages/CruiseForm'

const mapStateToProps = state => ({
    clients: getClientsSelector(state),
    cruise: getObjSelector(state),
    travelScript: travelGetObjSelector(state)
})

const mapDispatchToProps = {
    requestDelAction,
    requestGetSingleAction,
    requestSaveDataAction
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(CruiseForm)
