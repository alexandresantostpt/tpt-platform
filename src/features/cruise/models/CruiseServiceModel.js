import { config } from '@config'
import moment from 'moment'

import CruiseBoardingModel from './CruiseBoardingModel'
import CruiseLandingModel from './CruiseLandingModel'
import CruiseLibraryModel from './CruiseLibraryModel'

class CruiseServiceModel {
    constructor(
        {
            __v,
            _id,
            boarding,
            cabinNumber,
            cabinType,
            category,
            createdAt,
            landing,
            library,
            observation,
            passengers,
            reserveNumber,
            script,
            scriptDate,
            shipName,
            updatedAt
        },
        fromForm = false
    ) {
        this.__v = __v
        this._id = _id
        this._boarding = new CruiseBoardingModel(boarding, fromForm)
        this._cabinNumber = cabinNumber
        this._cabinType = cabinType
        this._category = category
        this._createdAt = moment(createdAt).utc(false)
        this._landing = new CruiseLandingModel(landing, fromForm)
        this._library = library ? new CruiseLibraryModel(library) : null
        this._passengers = passengers
        this._observation = observation
        this._reserveNumber = reserveNumber
        this._script = script
        this._scriptDate = scriptDate
        this._shipName = shipName
        this._updatedAt = moment(updatedAt).utc(false)
    }

    get v() {
        return this.__v
    }

    get id() {
        return this._id
    }

    get boarding() {
        return this._boarding
    }

    get cabinNumber() {
        return this._cabinNumber
    }

    get cabinType() {
        return this._cabinType
    }

    get category() {
        return this._category
    }

    get createdAt() {
        return this._createdAt
    }

    get landing() {
        return this._landing
    }

    get library() {
        return this._library
    }

    get observation() {
        return this._observation
    }

    get passengers() {
        return this._passengers
    }

    get reserveNumber() {
        return this._reserveNumber
    }

    get script() {
        return this._script
    }

    get scriptDate() {
        return this._scriptDate
    }

    get shipName() {
        return this._shipName
    }

    get updatedAt() {
        return this._updatedAt
    }

    getImage() {
        return this.library && this.library.image && `${config.api.url}/aerial/download/${this.library.image}`
    }

    /* eslint-disable-next-line complexity */
    toDTO() {
        return {
            __v: this.v,
            _id: this.id,
            boarding: this.boarding.toDTO() || null,
            cabinNumber: this.cabinNumber || null,
            cabinType: this.cabinType || null,
            category: this.category || null,
            landing: this.landing.toDTO() || null,
            library: this.library ? this.library.toDTO() : null,
            observation: this.observation || null,
            passengers: this.passengers || null,
            reserveNumber: this.reserveNumber || null,
            scriptDate: this.scriptDate || null,
            shipName: this.shipName || null
        }
    }

    toForm() {
        return {
            __v: this.v,
            _id: this.id,
            boarding: this.boarding.toForm(),
            cabinNumber: this.cabinNumber,
            cabinType: this.cabinType,
            category: this.category,
            landing: this.landing.toForm(),
            library: this.library ? this.library.toForm() : null,
            observation: this.observation,
            passengers: this.passengers,
            reserveNumber: this.reserveNumber,
            scriptDate: this.scriptDate,
            shipName: this.shipName
        }
    }
}

export default CruiseServiceModel
