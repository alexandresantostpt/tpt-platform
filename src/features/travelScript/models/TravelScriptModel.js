import moment from 'moment'

import Travel from '../../travel/TravelModel'
import TravelDateModel from './TravelDateModel'
import { isNotEmpty } from '@/utils/functions'

class TravelScriptModel {
    constructor({ __v, _id, dates, deleted, edited, travel, createdAt, updatedAt }) {
        this.__v = __v
        this._id = _id
        this._dates = dates && [...dates].map(date => new TravelDateModel(date))
        this._deleted = deleted
        this._edited = edited
        this._travel = new Travel(travel)
        this._createdAt = moment(createdAt).utc(false)
        this._updatedAt = moment(updatedAt).utc(false)
    }

    get v() {
        return this.__v
    }

    get id() {
        return this._id
    }

    get dates() {
        return this._dates
    }

    get deleted() {
        return this._deleted
    }

    get edited() {
        return this._edited
    }

    get travel() {
        return this._travel
    }

    get createdAt() {
        return this._createdAt
    }

    get updatedAt() {
        return this._updatedAt
    }

    hasServices() {
        return this.dates.some(date => date.services.length)
    }

    getRangeDates() {
        if (isNotEmpty(this._travel)) {
            return { endDate: this._travel.travelEndDate, startDate: this._travel.travelStartDate }
        }

        return {}
    }

    getClientsId() {
        return [...this.travel.clients].map(({ _id }) => _id)
    }

    toForm() {
        return {
            dates: this.dates,
            edited: this.edited,
            id: this.id,
            travel: this.travel,
            v: this.v
        }
    }
}

export default TravelScriptModel
