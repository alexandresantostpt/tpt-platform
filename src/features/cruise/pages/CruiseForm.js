import { events } from '@constants/events'

import PropTypes from 'prop-types'
import PubSub from 'pubsub-js'
import React, { useEffect, useRef, useState } from 'react'

import { colors } from '@helpers/colors'
import { getQueryParams } from '@utils/browser'
import { i18n } from '@i18n'
import { services } from '@/constants/services'
import { withRouter } from 'react-router-dom'

import Button from '@components/button/Button'
import Cancel from '@components/cancel/Cancel'
import FormBuilder from '@components/form/FormBuilder'
import Panel from '@components/panel/Panel'

import FormButtonContainer from '@components/form/container/button/FormButtonContainer'
import WrapperPadding from '@components/wrapper/WrapperPadding'

import { cruiseFormFields } from '../fields'

import CruiseServiceModel from '../models/CruiseServiceModel'

const CruiseForm = props => {
    const { cruise, match, travelScript } = props
    const {
        params: { cruiseId, id }
    } = match
    const { dateId, scriptDate, scriptId } = getQueryParams()

    const form = useRef()
    const previousPath = `/travel/${id}/script`

    const [image, updateImage] = useState()
    const [serviceDisabled, updateServiceDisabled] = useState(false)

    useEffect(() => {
        const { requestGetSingleAction } = props
        if (cruiseId) {
            requestGetSingleAction({ id: cruiseId })
            updateServiceDisabled(true)
        } else {
            requestGetSingleAction({ id: null })
            updateServiceDisabled(false)
        }
    }, [cruiseId])

    const callSaveData = values => {
        const { requestSaveDataAction } = props
        requestSaveDataAction({
            data: {
                dateId,
                scriptId,
                service: new CruiseServiceModel({ ...values, scriptDate }, true).toDTO(),
                type: services.CRUISE
            },
            image,
            redirectTo: previousPath
        })
        reset()
    }

    const onSaveClick = () => form.current.form.submit()

    const reset = () => {
        PubSub.publish(events.FORM_RESET)
        updateImage(null)
    }
    return (
        <>
            <WrapperPadding>
                <Panel backgroundImageUrl={cruise && cruise.getImage()} icon="ship" uploadImage={updateImage} />
            </WrapperPadding>
            <FormBuilder
                clients={props.clients}
                disabled={serviceDisabled}
                fields={cruiseFormFields}
                initialValues={cruise ? cruise.toForm() : { boarding: { date: scriptDate } }}
                onSubmit={callSaveData}
                rangeDates={travelScript && travelScript.getRangeDates()}
                ref={process.env.__TEST__ ? null : form}
            />
            <FormButtonContainer>
                <Button
                    color={colors.purple}
                    onClick={() => (serviceDisabled ? updateServiceDisabled(false) : onSaveClick())}
                    text={i18n.t(serviceDisabled ? 'buttons.editService' : 'buttons.saveService')}
                />
                <Cancel link={previousPath} />
            </FormButtonContainer>
        </>
    )
}

CruiseForm.defaultProps = {
    clients: [],
    cruise: {},
    travelScript: {}
}

CruiseForm.propTypes = {
    clients: PropTypes.array,
    cruise: PropTypes.objectOf(CruiseServiceModel),
    match: PropTypes.object.isRequired,
    requestDelAction: PropTypes.func.isRequired,
    requestGetSingleAction: PropTypes.func.isRequired,
    requestSaveDataAction: PropTypes.func.isRequired,
    travelScript: PropTypes.object
}

export default withRouter(CruiseForm)
