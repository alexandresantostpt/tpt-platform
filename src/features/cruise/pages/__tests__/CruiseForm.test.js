import React from 'react'
import renderer from 'react-test-renderer'

import CruiseServiceModel from '../../models/CruiseServiceModel'

import CruiseForm from '../CruiseForm'
import TravelScriptModel from '@/features/travelScript/models/TravelScriptModel'

describe('Tests for CruiseForm component', () => {
    const cruise = new CruiseServiceModel({
        __v: 0,
        _id: '5c52ebcd5fc7c9b1433ab704',
        boarding: {
            date: '2019-03-12T00:00:00.000Z',
            hour: '2019-03-12T20:00:00.458Z',
            point: 'Test'
        },
        cabinNumber: 123145,
        cabinType: 'Test',
        category: 'Test',
        cruiseName: 'Test',
        image: null,
        landing: {
            date: '2019-03-12T00:00:00.000Z',
            hour: '2019-03-12T20:00:00.458Z',
            point: 'Test'
        },
        observation: 'Observation',
        reserveNumber: 123456,
        script: '5c52ebcd5fc7c9b1433ab704',
        shipName: 'Princess Consuela'
    })
    const travelScript = new TravelScriptModel({ travel: { __v: 0 } })

    it('Should render withou crash', () => {
        const match = {
            params: {
                id: '5c52ebcd5fc7c949223ab705'
            }
        }
        const tree = renderer
            .create(
                <CruiseForm
                    cruise={cruise}
                    match={match}
                    requestDelAction={jest.fn()}
                    requestGetSingleAction={jest.fn()}
                    requestSaveDataAction={jest.fn()}
                    travelScript={travelScript}
                />
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
