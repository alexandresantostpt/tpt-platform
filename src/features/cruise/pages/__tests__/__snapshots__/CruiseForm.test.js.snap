// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for CruiseForm component Should render withou crash 1`] = `
Array [
  <div
    className="WrapperPadding-ovjbyx-0 lcnoja"
  >
    <div
      className="Panel__StyledPanel-sc-1qzkpb6-0 cGUlCn"
    >
      <span
        className="Panel__StyledIcon-sc-1qzkpb6-2 iavvSk"
      >
        <span
          className="icon icon-ship Icon__StyledIcon-crr1cg-0 dfhcZR"
          disabled={false}
          fontSize="3rem"
          onClick={null}
        />
      </span>
      <div
        className="UploadButton__StyledContainer-sc-1p8zoc-0 fpZFIG"
      >
        <label
          className="UploadButton__StyledUploadButton-sc-1p8zoc-3 ivhrYW"
          htmlFor="inputFile"
        >
          +
        </label>
        <input
          accept="image/png, image/jpeg"
          className="UploadButton__InputFile-sc-1p8zoc-4 eLaooA"
          id="inputFile"
          onChange={[Function]}
          type="file"
        />
      </div>
    </div>
    <ul
      className="PanelActions__StyledActions-sc-8xk8c8-0 lllkGw"
    />
  </div>,
  <form
    className="FormBuilder__StyledForm-c1htou-0 PZxQa"
    onSubmit={[Function]}
  >
    <div
      className="form__container"
    >
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Nome do cruzeiro
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-library.name"
          disabled={false}
          id="cruiseName"
          maxLength={120}
          name="library.name"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Nome do navio
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-shipName"
          disabled={false}
          id="shipName"
          maxLength={120}
          name="shipName"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="Princess Consuela"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Telefone de contato
          
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-library.phoneNumber"
          disabled={false}
          id="phoneNumber"
          maxLength={30}
          name="library.phoneNumber"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          type="text"
          value=""
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Site
          
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-library.site"
          disabled={false}
          id="site"
          maxLength={255}
          name="library.site"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          type="text"
          value=""
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Tipo de cabine
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-cabinType"
          disabled={false}
          id="cabinType"
          maxLength={30}
          name="cabinType"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="Test"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Categoria
          
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-category"
          disabled={false}
          id="category"
          maxLength={30}
          name="category"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          type="text"
          value="Test"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Número da cabine
          
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-cabinNumber"
          disabled={false}
          id="cabinNumber"
          maxLength={255}
          name="cabinNumber"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          type="text"
          value={123145}
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Código da reserva
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-reserveNumber"
          disabled={false}
          id="reserveNumber"
          maxLength={15}
          name="reserveNumber"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value={123456}
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Porto de embarque
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-boarding.point"
          disabled={false}
          id="boardingPoint"
          maxLength={255}
          name="boarding.point"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="Test"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Data de embarque
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormDate__StyledFormDate-sc-1cgz0rk-0 ebKpCI"
        >
          <div
            className="SingleDatePicker SingleDatePicker_1"
          >
            <div>
              <div
                className="SingleDatePickerInput SingleDatePickerInput_1"
              >
                <div
                  className="DateInput DateInput_1"
                >
                  <input
                    aria-describedby="DateInput__screen-reader-message-"
                    aria-label="Selecione uma data..."
                    autoComplete="off"
                    className="DateInput_input DateInput_input_1 DateInput_input__readOnly DateInput_input__readOnly_2"
                    disabled={false}
                    id=""
                    name=""
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Selecione uma data..."
                    readOnly={true}
                    required={false}
                    type="text"
                    value="12 Mar 2019"
                  />
                  <p
                    className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                    id="DateInput__screen-reader-message-"
                  >
                    Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                  </p>
                </div>
                <button
                  className="SingleDatePickerInput_calendarIcon SingleDatePickerInput_calendarIcon_1"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="icon icon-calendar Icon__StyledIcon-crr1cg-0 cmjVJQ"
                    disabled={false}
                    fontSize="1.75rem"
                    onClick={null}
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Hora de embarque
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormTime__StyledFormTime-sc-1ka3ing-1 bdjJPj"
        >
          <input
            className="FormTime__StyledDatePicker-sc-1ka3ing-0 hCpkqS"
            disabled={false}
            onChange={[Function]}
            options={
              Object {
                "blocks": Array [
                  2,
                  2,
                ],
                "delimiters": Array [
                  ":",
                ],
              }
            }
            placeholderText="Selecione um horário..."
            value="20:00"
          />
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Porto de desembarque
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-landing.point"
          disabled={false}
          id="landingPoint"
          maxLength={255}
          name="landing.point"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="Test"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Data de desembarque
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormDate__StyledFormDate-sc-1cgz0rk-0 ebKpCI"
        >
          <div
            className="SingleDatePicker SingleDatePicker_1"
          >
            <div>
              <div
                className="SingleDatePickerInput SingleDatePickerInput_1"
              >
                <div
                  className="DateInput DateInput_1"
                >
                  <input
                    aria-describedby="DateInput__screen-reader-message-"
                    aria-label="Selecione uma data..."
                    autoComplete="off"
                    className="DateInput_input DateInput_input_1 DateInput_input__readOnly DateInput_input__readOnly_2"
                    disabled={false}
                    id=""
                    name=""
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Selecione uma data..."
                    readOnly={true}
                    required={false}
                    type="text"
                    value="12 Mar 2019"
                  />
                  <p
                    className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                    id="DateInput__screen-reader-message-"
                  >
                    Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                  </p>
                </div>
                <button
                  className="SingleDatePickerInput_calendarIcon SingleDatePickerInput_calendarIcon_1"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="icon icon-calendar Icon__StyledIcon-crr1cg-0 cmjVJQ"
                    disabled={false}
                    fontSize="1.75rem"
                    onClick={null}
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Hora de desembarque
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormTime__StyledFormTime-sc-1ka3ing-1 bdjJPj"
        >
          <input
            className="FormTime__StyledDatePicker-sc-1ka3ing-0 hCpkqS"
            disabled={false}
            onChange={[Function]}
            options={
              Object {
                "blocks": Array [
                  2,
                  2,
                ],
                "delimiters": Array [
                  ":",
                ],
              }
            }
            placeholderText="Selecione um horário..."
            value="20:00"
          />
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Passageiros
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Observações
          
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-observation"
          disabled={false}
          id="observation"
          maxLength={255}
          name="observation"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          type="text"
          value="Observation"
        />
      </div>
    </div>
  </form>,
  <div
    className="FormButtonContainer-sc-1cqqivn-0 ADyLt"
  >
    <button
      className="Button__StyledButton-sc-1fc3klx-0 bEGBgE"
      disabled={false}
      fill={false}
      onClick={[Function]}
    >
      Salvar serviço
    </button>
    <a
      className="Cancel__StyledCancel-sc-1prgtit-0 WzuoR"
      to="/travel/5c52ebcd5fc7c949223ab705/script"
    >
      Cancelar
    </a>
  </div>,
]
`;
